### PR TITLE
Add missing classes in heading icon components

### PR DESCRIPTION
**Pull request by bartaz**
## Done

As part of preparation for typography updates I noticed quite a few inconsistencies in class names in heading icon component that were causing issues with upgrade.

This attempts to fix this by scoping down the CSS overrides added to `lazyloaded` div and adding all necessary class names for `p-heading-icon__img` and `p-heading-icon__title` elements that were missing across many instances of the component.

- Adds missing class names on elements in heading icon component
- Removes duplicated `__img` styles from `lazyloaded` div
- Removes unnecessary usage of heading icon component (where no icon was present)

## QA

Check all affected pages, make sure heading icon component is displayed as expected:

https://ubuntu-com-12807.demos.haus/0_docs/content_strips
https://ubuntu-com-12807.demos.haus/azure/sql
https://ubuntu-com-12807.demos.haus/blog/topics/robotics
https://ubuntu-com-12807.demos.haus/containers/shared/_whitepaper
https://ubuntu-com-12807.demos.haus/download/iot
https://ubuntu-com-12807.demos.haus/download/raspberry-pi
https://ubuntu-com-12807.demos.haus/download/raspberry-pi/thank-you
https://ubuntu-com-12807.demos.haus/edge-computing
https://ubuntu-com-12807.demos.haus/financial-services
https://ubuntu-com-12807.demos.haus/hpc
https://ubuntu-com-12807.demos.haus/internet-of-things/edgex
https://ubuntu-com-12807.demos.haus/internet-of-things/smart-city
https://ubuntu-com-12807.demos.haus/kubernetes/charmed-k8s
https://ubuntu-com-12807.demos.haus/kubernetes/compare
https://ubuntu-com-12807.demos.haus/kubernetes
https://ubuntu-com-12807.demos.haus/kubernetes/install
https://ubuntu-com-12807.demos.haus/kubernetes/managed
https://ubuntu-com-12807.demos.haus/kubernetes/what-is-kubernetes
https://ubuntu-com-12807.demos.haus/lxd
https://ubuntu-com-12807.demos.haus/managed/apps/kafka
https://ubuntu-com-12807.demos.haus/observability/managed
https://ubuntu-com-12807.demos.haus/openstack/consulting
https://ubuntu-com-12807.demos.haus/openstack/what-is-openstack
https://ubuntu-com-12807.demos.haus/security
https://ubuntu-com-12807.demos.haus/security/livepatch
https://ubuntu-com-12807.demos.haus/telco/osm/multi-cloud
https://ubuntu-com-12807.demos.haus/telco/osm/onboarding-network-functions

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

### CODE REVIEW

File: templates/download/raspberry-pi/index.html [hunk 1]
```diff
@@ -571,7 +571,7 @@ <h2>Make something great today</h2>
             height="150",
             hi_def=True,
             loading="lazy",
-            attrs={"class": "p-heading-icon__img--small"}
+            attrs={"class": "p-heading-icon__img p-heading-icon__img--small"}
```
Comment (anthonydillon) [inline templates/download/raspberry-pi/index.html:574]: This one looks better in production. Seems to align with the title better.

File: templates/download/raspberry-pi/thank-you.html [hunk 2]
```diff
@@ -144,7 +144,7 @@ <h2>Make something great today</h2>
             height="150",
             hi_def=True,
             loading="lazy",
-            attrs={"class": "p-heading-icon__img--small"}
+            attrs={"class": "p-heading-icon__img p-heading-icon__img--small"}
```
Comment (anthonydillon) [inline templates/download/raspberry-pi/thank-you.html:147]: This one seems better in production too

File: templates/kubernetes/compare.html [hunk 3]
```diff
@@ -372,6 +372,7 @@ <h3 class="u-sv2">Kubernetes resources</h3>
   </div>
   <div class="row p-divider">
     <div class="col-4 p-divider__block">
+      <!-- TODO: add missing img and title classes -->
```
Comment (anthonydillon) [inline templates/kubernetes/compare.html:375]: Missing image? I see a datasheet one below.

File: templates/openstack/what-is-openstack.html [hunk 4]
```diff
@@ -666,7 +666,9 @@ <h3>Learn more about OpenStack</h3>
               width="171",
               height="150",
               hi_def=True,
-              loading="lazy" ) |
+              loading="lazy",
+              attrs={"class": "p-heading-icon__img"},
+              ) |
               safe }}
             </div>
             <h4 class="p-heading-icon__title" style="color: #fff">
```
Comment (anthonydillon) [inline templates/openstack/what-is-openstack.html:674]: With the `is-dark` do we need the online style here? Could you change the text to sentence case as a drive by?

File: templates/telco/osm/onboarding-network-functions.html [hunk 5]
```diff
@@ -188,11 +188,9 @@ <h3 class="p-heading--4">OSM in KubeCon 2021, LA</h3>
...
   <div class="row">
     <div class="col-12">
-      <div class="p-heading-icon--small">
-        <h3 class="p-heading--4" id="tell-us">Tell us about your network function</h3>
-        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
-        <p><a href="https://charmed-osm.com/#tell-us" class="p-button--positive">Start now</a></p>
-      </div>
+      <h3 class="p-heading--4" id="tell-us">Tell us about your network function</h3>
+      <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+      <p><a href="https://charmed-osm.com/#tell-us" class="p-button--positive">Start now</a></p>
```
Comment (anthonydillon) [inline templates/telco/osm/onboarding-network-functions.html:193]: This link should be: https://charmed-osm.com/contact-us

**Review by anthonydillon — APPROVED** [review]
Thanks @bartaz, good work 👍. Generally looks good, just a few suggestions in the code.

File: templates/download/raspberry-pi/index.html (hunk 1 above)
Comment (bartaz) [reply templates/download/raspberry-pi/index.html:574]: I know, but it needs the `p-heading-icon__img` classname that comes from Vanilla, `p-heading__img--small` is custom ubuntu.com class, and for reliable migration we can't rely just on it.

I'll see what I can do to improve the look.

File: templates/kubernetes/compare.html (hunk 3 above)
Comment (bartaz) [reply templates/kubernetes/compare.html:375]: Nevermind, this was a TODO comment for myself to know which classnames are missing, I forgot to remove it.

File: templates/download/raspberry-pi/index.html (hunk 1 above)
Comment (bartaz) [reply templates/download/raspberry-pi/index.html:574]: Just checked, and I don't see much difference TBH.

Prod:
image

Branch:

image

File: templates/kubernetes/compare.html (hunk 3 above)
Comment (bartaz) [reply templates/kubernetes/compare.html:375]:
```suggestion
```

File: templates/openstack/what-is-openstack.html (hunk 4 above)
Comment (bartaz) [reply templates/openstack/what-is-openstack.html:674]: There seem to be couple layers of custom styles that override `is-dark` colour from strip. I don't want to mess up too deeply with ubuntu.com styling.

File: templates/telco/osm/onboarding-network-functions.html (hunk 5 above)
Comment (bartaz) [reply templates/telco/osm/onboarding-network-functions.html:193]:
```suggestion
      <p><a href="https://charmed-osm.com/contact-us" class="p-button--positive">Start now</a></p>
```

**Pull request merged**